manager/controllers/app/blueprint_controller.go

```diff
@@ -215,11 +197,33 @@ func (r *BlueprintReconciler) applyChartResource(ctx context.Context, log zerolo
 			}
 		}
 	}
+	return registrySuccessfulLogin, nil
+}
+
+func (r *BlueprintReconciler) applyChartResource(ctx context.Context, log *zerolog.Logger, chartSpec fapp.ChartSpec,
+	args map[string]interface{}, blueprint *fapp.Blueprint, releaseName string) (ctrl.Result, error) {
+	log.Trace().Str(logging.ACTION, logging.CREATE).Msg("--- Chart Ref ---\n\n" + chartSpec.Name + "\n\n")
+	kubeNamespace := blueprint.Spec.ModulesNamespace
+
+	args = CopyMap(args)
+	for k, v := range chartSpec.Values {
+		SetMapField(args, k, v)
+	}
+	nbytes, _ := yaml.Marshal(args)
+	log.Trace().Str(logging.ACTION, logging.CREATE).Msg("--- Values.yaml ---\n\n" + string(nbytes) + "\n\n")
+
+	var registrySuccessfulLogin string
+	var err error
+	registrySuccessfulLogin, err = r.obtainSecrets(ctx, log, chartSpec)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	tmpDir, err := ioutil.TempDir("", "fybrik-helm-")
 	if err != nil {
 		return ctrl.Result{}, errors.WithMessage(err, chartSpec.Name+": failed to create temporary directory for chart pull")
 	}
-	defer func(log zerolog.Logger) {
+	defer func(log *zerolog.Logger) {
 		if err = os.RemoveAll(tmpDir); err != nil {
 			log.Error().Msg("Error while calling RemoveAll on directory created for pulling helm chart")
```

Comment:
maybe worthwhile to add the tmpDir name into the error message

Reply:
Added.